pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -276,7 +272,20 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	conditions := r.getAvailableConditions()
+	conditions, err := r.getMCPConditionsByProfile(instance)
+	if err != nil {
+		conditions := r.getDegradedConditions(conditionReasonComponentsCreationFailed, err.Error())
```

Comment:
please use a new reason, components are created successfully at this point.

Reply:
We can still get an MCP degraded state after components were successfully created.
I tried keeping this as idempotent as I can.
Basically we will get new requests (new reconcile loops) on every mcp status change (mcp that relates to an existing Performance Profile) 
So if an existing on is re-passed here due to a MCP status change: the components shouldn't re-apply  and the available conditions will remain in case no degraded state is detected

Reply:
it's just about the reason in the condition (`conditionReasonComponentsCreationFailed`). It should be something like `FailedGettingMCPStatus`). Or do I misunderstand something?

Reply:
right. missed that extra redundant line. it shouldn't have been there.
good catch.